src/mono/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeLocalBuilder.Mono.cs

```diff
@@ -99,5 +99,8 @@ internal int EndOffset
         {
             get { return endOffset; }
         }
+
+        // Mono uses ILGenerator instance instead of MethodBuilder, could not get reference to the MethodBuilder the local belongs to.
+        public override MethodInfo Method => throw new NotImplementedException();
```

Comment:
This looks suspect.

Do we really need to expose `Method` as a public abstract property? Would it be enough to expose it as an internal method on LocalBuilderImpl?

Reply:
I thought it will be OK as mono is not using it anywhere ...
> Would it be enough to expose it as an internal method on LocalBuilderImpl?

Yes, it can be an internal method, though would need to cast from `LocalBuilder` into `LocalBuilderImpl`, thought adding a public API could be useful if someone build a new `ILGenerator` implementation ...

Reply:
This is a new public API. It needs to work on all runtimes.

If somebody build a new ILGenerator, I think they can keep track of the method just like we do in our own ILGeneratorImpl.

Reply:
@lambdageek for suggestions here.

Reply:
@buyaa-n

1. You probably know more at this point, but it doesnt' seem like it would be a problem to add a `Method` member to mono's `RuntimeILGenerator` and add an extra argument to all the `RuntimeILGenerator` constructors.

2. Actually I have a question about this API: Why is it a `MethodInfo` and not a `MethodBase` ? Constructors can have locals too.